### DISCUSSION TITLE
Can results grid group messages by correlation ID or session ID?

### DISCUSSION THREAD

**Discussion by pwakano** [sipcapture/homer#478] [Q&A]
Hi there,
Hope you all doing well!
I was analysing the possibilities of the Sipcature tool and was wondering if it is possible to group the messages in the result grid by the session ID or the correlation ID.
The result grid already apply different colours to the each session ID for example, so, is there any way to group them so all messages get collapsed into only 1 line?
Kind regards,
Thank you,
Patrick Wakano

**Comment by Gasmanz** [comment]
I agree. It seems kind of redundant to show for example multiple SIP INVITE messages for a Session/Call ID as when you click on any of the Session ID lines they always open to the SIP flow, and you can then click on any of the SIP INVITE messages to see their details. My best guess would be that when searching for a SIP call/session most people are firstly looking for the session flow anyway, and not interested in direct access to all the Session INVITE messages from the Search Results page. This would be great as option if not the default.

**Reply by pwakano** [reply]
That's exactly my feeling/experience.

**Comment by lmangani** [comment]
You can click on the SIP Method from the results to see the individual message, or click the Call-ID for correlation. That's the design.
If you only want to get INVITEs, there's a METHOD filter that allows exactly that type of behaviour out of the box. Options are there.

**Reply by pwakano** [reply]
Sorry, I am afraid there is a misunderstanding.... The in-dialog requests take up space in the UI and because we are usually searching for calls, not a specific request, I was asking if there was a way to group all the requests of the same session ID under one line. This would allow more calls to be displayed on the page at once.

**Reply by Gasmanz** [reply]
I agree. In every search I have done I'm looking to find a specific call (from number, to number and approx start time) as a result and not all the invites and re-invites, or all the 200 OK responses. I don't know of any SIP engineers that want to have multiple matches for the same call in the search results view. It's basically too noisy with all that info showing and means you display less unique call matches per page. For example here is a call with 8 invite/re-invite messages.

![image](https://user-images.githubusercontent.com/81398328/152087805-a5ed31d0-bbde-4543-a1cc-d599a0aff7f7.png)

The first invite message would be enough to identify the call I'm looking for, and I could look at all the other invite messages in the flow if I needed to.

**Reply by pwakano** [reply]
Specially true on a call center env where the TO and FROM numbers are the same very frequently.

**Reply by NiteCrwlr** [reply]
THIS function would be NICE! I think we only need one row per call (for example date, session-id, from, to), if I need more deailed info and the call flow, I can click on it.

**Reply by sudoritz** [reply]
is this added in future version since this Q/A post ?
